### PR TITLE
Fix plural rule detection for languages

### LINKED ISSUE #14627: Chinese plural rule is broken
```
Unhandled promise rejection TypeError: "e is undefined"

    translate l10n.js:108
translate l10n.js:107
translatePlural l10n.js:156
…
update filesummary.js:224
setSummary filesummary.js:190
calculate filesummary.js:173
setFiles filelist.js:1311
reloadCallback filelist.js:2014
…
```

Also the summary `%n files` shows `和` instead of being replaced with the number of files. Deleting all plural rules from all zh_CN.js files removes the error.
adding a second item into the arrays also works, so it seems that the plural rule somehow tries to use a wrong plural rule, because chinese only has 1 rule.

Maybe @rullzer you can help? I tried debugging it, but I fail to recompile the new file after js/core/l10n.js was modified.

It also breaks worse in other apps, when the plural string is required instead of being added afterwards: https://github.com/nextcloud/announcementcenter/issues/146

### PR DESCRIPTION
**Pull request by nickvergessen**
Plurals depend on the language, not the locale :thinking: 

Regression from #11028 
Fix #14627

### CODE REVIEW

File: settings/js/vue-1.js.map [hunk 1]
```diff
@@ -1 +1 @@
...quire('vue'))\n  if (api.compatible) {\n    module.hot.accept()\n    if (!module.hot.data) {\n      api.createRecord('257430ef', component.options)\n    } else {\n      api.reload('257430ef', component.options)\n    }\n    \n  }\n}\ncomponent.options.__file = \"settings/src/components/prefixMixin.vue\"\nexport default component.exports","import mod from \"-!../../../node_modules/babel-loader/lib/index.js!../../../node_modules/vue-loader/lib/index.js??vue-loader-options!./prefixMixin.vue?vue&type=script&lang=js&\"; export default mod; export * from \"-!../../../node_modules/babel-loader/lib/index.js!../../../node_modules/vue-loader/lib/index.js??vue-loader-options!./prefixMixin.vue?vue&type=script&lang=js&\"","var render, staticRenderFns\nimport script from \"./svgFilterMixin.vue?vue&type=script&lang=js&\"\nexport * from \"./svgFilterMixin.vue?vue&type=script&lang=js&\"\n\n\n/* normalize component */\nimport normalizer from \"!../../../node_modules/vue-loader/lib/runtime/componentNormalizer.js\"\nvar component = normalizer(\n  script,\n  render,\n  staticRenderFns,\n  false,\n  null,\n  null,\n  null\n  \n)\n\n/* hot reload */\nif (module.hot) {\n  var api = require(\"/home/admin/Do
```
Comment (nickvergessen) [inline settings/js/vue-1.js.map]: require(\"/home/admin/Docker/server/node_modules/

vs.
require(\"/home/nickv/…/server/node_modules/

can we get rid of this somehow?

Comment (ChristophWurst) [reply settings/js/vue-1.js.map]: @juliushaertl @rullzer @skjnldsv we discussed that before. This was something about webpack hot reloading. Didn't we fix that?

File: core/src/OC/l10n.js [hunk 2]
```diff
@@ -173,7 +173,7 @@ const L10n = {
 	 * @private
 	 */
 	_getPlural: function(number) {
-		var locale = OC.getLocale();
+		var locale = OC.getLanguage();
```
Comment (ChristophWurst) [inline core/src/OC/l10n.js]: would it make sense to rename the variable, just to make it less confusing?

File: settings/js/vue-1.js.map (hunk 1 above)
Comment (ChristophWurst) [reply settings/js/vue-1.js.map]: This should be fixed as of https://github.com/nextcloud/server/pull/14258.

Comment (ChristophWurst) [reply settings/js/vue-1.js.map]: Actually I don't get why the settings bundle even changed for you. Did you `npm i`? Is there an old `node_modules` in your `settings`?

Comment (nickvergessen) [reply settings/js/vue-1.js.map]: I ran `make dev-setup` and then `make build-js-production`. If root builds all,it should also reset all node_modules?

Comment (rullzer) [reply settings/js/vue-1.js.map]: actually it only checks your root folder for node modules. But there could be old leftovers. do you have a node_modules folders in settings? If so. remove it.

File: core/src/OC/l10n.js (hunk 2 above)
Comment (MorrisJobke) [reply core/src/OC/l10n.js]: Makes sense to rename `locale` to `language` as well.

**Comment by nickvergessen** [comment]
renamed the variable

**Review by MorrisJobke — APPROVED** [review]
Code looks good and works 👍

File: settings/js/vue-1.js.map (hunk 1 above)
Comment (nickvergessen) [reply settings/js/vue-1.js.map]: No, there is none, also I recompiled and it always does this change...

**Comment by MorrisJobke** [comment]
Retriggered CI because the JSUnit failure looks weird.

**Review by skjnldsv — APPROVED** [review]

**Comment by MorrisJobke** [comment]
JSUnit needs to be fixed :/

**Comment by MorrisJobke** [comment]
stable15 in #14867

**Pull request merged**